changelog/6693.feature.rest

```diff
@@ -0,0 +1,3 @@
+Able to detect gzip-compressed FITS files even if they don't have the .gz extension in filename.
+`~sunpy.io.file_tools.detect_filetype` now looks for the right file signature while checking
```

Comment:
Should fix the documentation build:
```suggestion
`~sunpy.io.detect_filetype` now looks for the right file signature while checking
```
https://docs.sunpy.org/en/latest/generated/api/sunpy.io.detect_filetype.html#detect-filetype